docs/source/dataframe.rst

```diff
@@ -128,48 +150,53 @@ The following class of computations works well:
     *  Drop duplicates:  ``df.x.drop_duplicates()``
     *  Join on index:  ``dd.merge(df1, df2, left_index=True, right_index=True)``
        or ``dd.merge(df1, df2, on=['idx', 'x'])`` where ``idx`` is the index
        name for both ``df1`` and ``df2``
-    *  Join with Pandas DataFrames: ``dd.merge(df1, df2, on='id')``
+    *  Join with pandas DataFrames: ``dd.merge(df1, df2, on='id')``
     *  Element-wise operations with different partitions / divisions: ``df1.x + df2.y``
     *  Date time resampling: ``df.resample(...)``
     *  Rolling averages:  ``df.rolling(...)``
     *  Pearson's correlation: ``df[['col1', 'col2']].corr()``
-* Operations requiring a shuffle (slow-ish, unless on index)
+* Operations requiring a shuffle (slow-ish, unless on index, see :doc:`dataframe-groupby`)
     *  Set index:  ``df.set_index(df.x)``
     *  groupby-apply not on index (with anything):  ``df.groupby(df.x).apply(myfunc)``
     *  Join not on the index:  ``dd.merge(df1, df2, on='name')``
 
-However, Dask DataFrame does not implement the entire Pandas interface.  Users
+However, Dask DataFrame does not implement the entire pandas interface.  Users
 expecting this will be disappointed.  Notably, Dask DataFrame has the following
 limitations:
 
 1.  Setting a new index from an unsorted column is expensive
 2.  Many operations like groupby-apply and join on unsorted columns require
     setting the index, which as mentioned above, is expensive
-3.  The Pandas API is very large.  Dask DataFrame does not attempt to implement
-    many Pandas features or any of the more exotic data structures like NDFrames
-4.  Operations that were slow on Pandas, like iterating through row-by-row,
+3.  The pandas API is very large.  Dask DataFrame does not attempt to implement
+    many pandas features or any of the more exotic data structures like NDFrames
+4.  Operations that were slow on pandas, like iterating through row-by-row,
     remain slow on Dask DataFrame
 
-See :doc:`DataFrame API documentation<dataframe-api>` for a more extensive list.
+See the :doc:`DataFrame API documentation<dataframe-api>` for a more extensive list.
 
 
 Execution
 ---------
 
-By default, Dask DataFrame uses the multi-threaded scheduler.
-This exposes some parallelism when Pandas or the underlying NumPy operations
-release the global interpreter lock (GIL).  Generally, Pandas is more GIL
+By default, Dask DataFrame uses the :ref:`multi-threaded scheduler <threaded-scheduler>`.
+This exposes some parallelism when pandas or the underlying NumPy operations
+release the global interpreter lock (GIL).  Generally, pandas is more GIL
 bound than NumPy, so multi-core speed-ups are not as pronounced for
-Dask DataFrame as they are for Dask Array.  This is changing, and
-the Pandas development team is actively working on releasing the GIL.
+Dask DataFrame as they are for Dask Array. This is particularly true
+for string-heavy Python DataFrames, as Python strings are GIL bound.
```

Comment:
@scharlottej13 I just want to confirm if "Python" DataFrames is intentional because I haven't seen this vocabulary used before

Reply:
Yes! Good question, "python dataframe" is a popular keyword search, so I was looking for an opportunity to use it somewhere in this page.

Reply:
Got it! Thanks!

Reply:
No worries at all :)